src/en/guide/upgrading/upgrading_4_0_x.adoc

```diff
@@ -0,0 +1,80 @@
+### Bump up Grails Version
+
+You will need to upgrade your Grails version defined in `gradle.properties` as:
+
+[source,groovy,subs="attributes"]
+.gradle.properties
+----
+...
+grailsVersion=4.1.0.M1
+...
+----
+
+### Groovy 3.0.3
+
+Grails 4.1 provide support for Groovy 3. We would recommend you to please check the https://groovy-lang.org/releasenotes/groovy-3.0.html[Release notes for Groovy 3] to update your application in case you are using a specific feature which might not work in Groovy 3.
+
+Define groovyVersion in `gradle.properties` to force the application to use Groovy 3.
```

Comment:
What does this mean? Why would I need to "force" my app to use groovy 3 if grails 4.1 is using groovy 3?

Reply:
@erichelgeson Basically the groovy-all nonsense and conflicting transitive dependencies. Setting the property just sets up the resolution strategy to force the supplied version

Reply:
Could Grails set the default (eg for 4.1.0 its 3.0.3, but for 4.1.2 it could be 3.0.5) - the common use case would be to take the grails defaults, the uncommon would be to override manually. Feels awkward to require the user to set it.